components/raftstore/src/store/peer.rs

```diff
@@ -4211,8 +4212,23 @@ where
             if entry.get_data().is_empty() {
                 continue;
             }
-            let cmd: RaftCmdRequest =
-                util::parse_data_at(entry.get_data(), entry.get_index(), &self.tag);
+
+            let cmd = if ctx.cfg.enable_v2_compatible_learner {
```

Comment:
Why we handle in pre_propose_prepare_merge? Do we need to handle it elsewhere like pre_propose?

Reply:
Not need for proxy. Just change it to make tikv raftstore v1 possibly works with v2. And there is no other place.